interface/wx/memoryallocator.h

```diff
@@ -0,0 +1,98 @@
+/////////////////////////////////////////////////////////////////////////////
+// Name:        memoryallocator.h
+// Purpose:     interface of wxMemoryAllocator
+// Author:      wxWidgets team
+// Licence:     wxWindows licence
+/////////////////////////////////////////////////////////////////////////////
+
+/**
+    @class wxMemoryAllocator
+
+    This is a little helper class that hides the difference between
+    C++ operators new/delete and C way of memory management using
+    malloc()/free(). It is currently being used in wxImage to do
+    memory management. Right now it has 3 direct descendants - 1
+    for C-way, 1 for C++-way and 1 that will not throw on allocation.
+
+    @since 3.1.1
+*/
+class wxMemoryAllocator
+{
+public:
+    /**
+        Destructor.
+    */
+    virtual ~wxMemoryAllocator();
+    
+    /**
+        Default memory allocation function. This function may
```

Comment:
The method in the base class is pure and hence does nothing and its descriptions should reflect that?

Whether the function on failure returns NULL or throws should be stated in method descriptions in the derived classes.